### DISCUSSION TITLE
Maps integration for Point fields

### DISCUSSION THREAD

**Discussion by ahmedHusseinF** [payloadcms/payload#1258] [Feature Requests & Ideas]
Lately I have been fiddling around in payload and thought it was a better admin UX point is allowing the admin to pick the lat/lng of a Point field using an actual Map (Google/OpenStreetMaps).

I am proposing having 2 sets of components out of the box for integrating the Point field with Google Maps or OpenStreetMaps and allow the user to configure both from the Point field config right in the model.

Some of the options for the components can be:
1. Height (both)
2. Width (both)
3. API Keys (Google Maps)
4. Tile Layers (OpenStreetMaps)
5. Possible Event Handlers (both)

**Comment by ahmedHusseinF** [comment]
A proof of concept for integrating the Point field with OpenStreetMaps can be seen in the repo [ahmedHusseinF/payload-leaflet](https://github.com/ahmedHusseinF/payload-leaflet)

**Reply by JarrodMFlesch** [reply]
I must have missed this comment (or we were writing replies at the same time earlier today), I think this is the right approach. I think making your own custom field is what you should do, I'm not certain it belongs in the core product.

Cool custom field 😃

**Reply by ahmedHusseinF** [reply]
I meant a set of custom components (Field/Cell/Filter).
For each maps provider, a set of components will be exported.

**Comment by JarrodMFlesch** [comment]
I think this could make an awesome custom field! I don't think payload wants to assume you are going to use a specific map provider in order to use the point field.

But as a custom field I think this could definitely provide value for others looking to do what you are talking about. 

If you end up giving this a shot, we are around to help and provide feedback/guidance 👍

**Comment by Rar9** [comment]
+1

Adding Mapbox api / Google maps api would allow the users to choose accord to their preference.

**Reply by ahmedHusseinF** [reply]
We can agree on base providers like for sure Google & OpenStreetMaps.
Adding more will be better DX for sure and I am all for it.

**Comment by Velua** [comment]
How did this go? Did anything get developed?

**Reply by tyteen4a03** [reply]
Nope, but I think a plugin would be cool!
